src/main/java/com/google/api/codegen/config/PageStreamingConfig.java

```diff
@@ -237,10 +240,16 @@ static PageStreamingConfig createPageStreamingConfig(
 
     // Toggle pagination based on presence of paging params.
     // See https://cloud.google.com/apis/design/design_patterns for API pagination pattern.
-    ProtoField tokenField = methodModel.getInputField(ProtoPagingParameters.nameForPageToken());
-    ProtoField pageSizeField = methodModel.getInputField(ProtoPagingParameters.nameForPageSize());
+    ProtoPagingParameters pagingParams = new ProtoPagingParameters();
+    ProtoField tokenField = methodModel.getInputField(pagingParams.getNameForPageToken());
+    ProtoField pageSizeField = methodModel.getInputField(pagingParams.getNameForPageSize());
+    if (pageSizeField == null) {
+      if (language == TargetLanguage.JAVA && transportProtocol == TransportProtocol.HTTP) {
```

Comment:
The generators cannot not know which client they generated. The number of places where it is needed is limited, but is still present. It is easy for me to remove this check, but I'm afraid of breaking some of the existing clients OnePlatform clients

Reply:
Added.
I noticed you alsmost always ask for reformulating my original comments/todos. I don't have a strong opinion about wording of these things. For those comments which you have strong opinion about please consider providing the exact wording you want it to be in the original review comment, this will let us save some review cycles.